### PR TITLE
Make TF Eager the default runtime.

### PR DESCRIPTION
**Pull request by bgogul**
We need this for the jupyter notebooks and this is the future anyway. :)

### CODE REVIEW

**Comment by bgogul** [comment]
@swift-ci please test tensorflow

File: stdlib/public/TensorFlow/CompilerRuntime.swift [hunk 1]
```diff
@@ -111,7 +111,7 @@ public enum _RuntimeConfig {
 
   /// When true, uses the TF eager C API, and TF interpreter backend.
   /// Otherwise uses the TF C API, with execution mode set below.
```
Comment (rxwei) [inline stdlib/public/TensorFlow/CompilerRuntime.swift]: It'd be nice to add a note here to say that sends/receives for variant and resources tensors are not supported and have undefined behavior when this flag is turned off.

**Review by rxwei — APPROVED** [review]
LGTM. This should've been enabled in #19636 because `--tf-send-recv-opaque-handle` will not work without the eager runtime. You can also remove [this line](https://github.com/bgogul/swift/blob/389e377b639684affb777ba71559169a36a01253/test/TensorFlowRuntime/dataset_global.swift#L23) in tests which turns on `usesTFEagerAPI` explicitly.

**Review by mhong — APPROVED** [review]
Makes sense. Thanks for changing this!

File: stdlib/public/TensorFlow/CompilerRuntime.swift (hunk 1 above)
Comment (bgogul) [reply stdlib/public/TensorFlow/CompilerRuntime.swift]: Done.

**Comment by bgogul** [comment]
>  You can also remove this line in tests which turns on usesTFEagerAPI explicitly.
This is done as well. 

I will merge this after https://github.com/apple/swift/pull/19865 that disables the test.

File: stdlib/public/TensorFlow/CompilerRuntime.swift [hunk 2]
```diff
@@ -111,7 +111,9 @@ public enum _RuntimeConfig {
...
-  static public var usesTFEagerAPI = false
+  /// NOTE: when this is false, sends/receives for variant and resources tensors
```
[the remark this answers is not in the document]
Comment (bgogul) [reply stdlib/public/TensorFlow/CompilerRuntime.swift]: Good to know. :)  Done.

**Pull request merged**